awx/main/models/credential.py

```diff
@@ -689,6 +689,16 @@ def vault(cls):
                 'type': 'string',
                 'secret': True,
                 'ask_at_runtime': True
+            }, {
+                'id': 'vault_id',
+                'label': 'Vault Identifier',
+                'type': 'string',
+                'format': 'vault_id',
+                'help_text': ('Specify an (optional) Vault ID. This is '
```

Comment:
👍 
Adding the vault id as an optional per-credential field makes sense to me - my understanding is that we'll be picking up which vault id's correspond to passwords that need prompting using the formatting proposed here: https://github.com/ansible/awx/issues/352#issuecomment-337995178 which should be enough information to cue the UI on which individual vault credentials need prompting.

Reply:
@jakemcdermott yep, that's exactly right; I plan on having the launch endpoint expose e.g.,

```json
{
    "passwords_needed_to_start": [
        "vault_password.id1",
        "vault_password.id2"
    ]
}
```